artifact-manifest-spec.md

```diff
@@ -0,0 +1,76 @@
+# OCI Artifact Manifest Spec (Phase-1 Reference Types)
+The OCI artifact manifest generalizes the use cases of [OCI image manifest][oci-image-manifest-spec] by removing constraints defined on the image-manifest such as a required `config` object and required & ordinal `layers`. It then adds a `subjectManifest` property supporting reference types. The addition of a new manifest does not change, nor impact the `image.manifest`. It provides a means to define a wide range of artifacts, including a chain of related artifacts enabling SBoMs, on-demand loading, signatures and metadata that can be related to an `image.manifest` or `image.index`. By defining a new manifest, registries and clients opt-into new capabilities, without breaking existing registry and client behavior or setting expectations for scenarios to function when the client and/or registry doesn't yet implement the new capabilities.
+
+To enable a fall 2021 focus on supply chain security,  **Phase 1** will narrowly focus on Reference Type support, giving time for further generalization with less time constraints.
+
+For usage and scenarios, see [artifact-manifest.md](./artifact-manifest.md)
+
+## Example OCI Artifact Manifests
+
+The following are Phase 1 examples:
+
+- [`net-monitor:v1` oci container image](./artifact-manifest/net-monitor-oci-image.json)
+- [`net-monitor:v1` notary v2 signature](./artifact-manifest/net-monitor-image-signature.json)
+- [`net-monitor:v1` sample sbom](./artifact-manifest/net-monitor-image-sbom.json)
+- [`net-monitor:v1` nydus image with on-demand loading](./artifact-manifest/net-monitor-image-nydus-ondemand-loading.json)
+
+## OCI Artifact Manifest Properties
+
+For **Phase 1**, an artifact manifest provides an optional collection of blobs and a reference to the manifest of another artifact.
+
+- **`schemaVersion`** *int*
```

Comment:
I would just drop this, as it is a vestige of old types. No need to carry this into new versions.

Reply:
can I get a hell yes!

Reply:
I've always wondered how `schemaVersion` relates to the embedded `mediaType` version: `application/vnd.oci.artifact.manifest.v1+json` 
I'd be happy to avoid the duplication.

Reply:
All of everything we see today is `schemaVersion` 2. It was just used to differentiate it early on. It's not needed for new manifests where this doesn't matter.

Reply:
Just asking for clarity. 
There is a difference in the schema of the .json document between `image.manifest`, `image.index` and `artifact.manifest`
What I believe you're saying is the `manifest.mediaType` version provides enough info for how to process the `.json` document?

Reply:
> What I believe you're saying is the `manifest.mediaType` version provides enough info for how to process the `.json` document?

Implementations _generally_ shouldn't be trying to peek inside an artifact in order to understand how to process it. By that point, it's often too late. Instead, content should _generally_ be presented alongside a descriptor, which contains the mediaType of the content.

For a registry, these descriptors are sometimes communicated via HTTP headers by the client on push and the server on pull. Sometimes, the descriptors are embedded within other content, e.g. when pulling a manifest list, you select one of the `manifests` entries (based on some criteria, e.g. `platform`) and process what you fetch based on the mediaType in that descriptor.

If you try to use the embedded mediaType to process everything, you'll probably end up having to parse content twice. Once to find its mediaType, then again to parse based on the mediaType.

Very old clients didn't send or process HTTP headers sufficiently to behave properly, so this schemaVersion existed to differentiate between schema 1 and schema 2 images. Now that we live in a universe with a nice type system, that shouldn't be necessary.

If you _were_ to add a schemaVersion field to a new kind of artifact, you absolutely don't want to set it to `1` or `2`, because this could break old clients that are using schemaVersion to select between two manifest schemas. You _could_ use `schemaVersion: 3`, but it's easier to just drop it altogether and rely on mediaTypes and descriptors in order to parse content.

Reply:
Thanks Jon,
Based on all this feedback and more understanding for how versioning is processed for the document to safely evolve, I did remove `schemaVersion` from the current [artifact.manifest](https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest-spec.md) and [examples](https://github.com/oras-project/artifacts-spec/tree/main/artifact-manifest)